packages/api/src/controllers/stream.ts

```diff
@@ -583,21 +593,19 @@ app.get("/:parentId/sessions", authorizer({}), async (req, res) => {
   const raw = req.query.raw && req.user.admin;
 
   const stream = await db.stream.get(parentId);
-  if (
-    !stream ||
-    (stream.deleted && !req.isUIAdmin) ||
-    (stream.userId !== req.user.id && !req.isUIAdmin)
-  ) {
-    res.status(404);
-    return res.json({ errors: ["not found"] });
-  }
+  req.checkResourceAccess(stream, true);
 
   let filterOut;
   const query = [];
   query.push(sql`data->>'parentId' = ${stream.id}`);
   query.push(sql`(data->'lastSeen')::bigint > 0`);
   query.push(sql`(data->'sourceSegmentsDuration')::bigint > 0`);
   query.push(sql`data->>'partialSession' IS NULL`);
+  if (!req.isUIAdmin) {
+    query.push(
+      sql`coalesce(data->>'projectId', '') = ${req.project?.id || ""}`
+    );
+  }
```

Comment:
I don't think this is required, there's no `userId` check here either right? Probably because child streams are necessarily from the same user (and also from the same project) as their parent.

I'd say we follow the rule of thumb I suggested and only add project checks on the same place where we have user checks. No need to filter by project here if we're not filtering by user.

If you prefer to keep this filter, WDYT of adding the corresponding user filter here as well? I think it's confusing to check only 1 of the 2, especially if only the project.